### PR TITLE
CO-72: Ensure tasks that shell out are resilient to non-existent binaries

### PR DESCRIPTION
**Pull request by carlpulley**
- [x] Define command checks
- [x] Place command checks near usage sites
- [x] Add in SBT plugin tests for the sbt-cake task `checkExternalBuildTools`
---
- [x] Ready for merge (depends on PR #10)
- [x] Ready for review

### CODE REVIEW

**Comment by fommil** [comment]
`scripted` tests are always :100:

File: src/sbt-test/sbt-cake/simple/src/main/scala/Hello.scala [hunk 1]
```diff
@@ -1 +1 @@
-object TerribleFormatting { def foo = "bar" }
+object TerribleFormatting{def foo=   "bar"}
```
Comment (amirkarimi) [inline src/sbt-test/sbt-cake/simple/src/main/scala/Hello.scala:1]: Isn't it better to add some comment on top of this code explaining that this is malformatted intentionally?

Comment (fommil) [reply src/sbt-test/sbt-cake/simple/src/main/scala/Hello.scala:1]: if you feel that people need a comment on a badly formatted piece of code, literally called `TerribleFormatting`, then by all means go ahead and add one.

Comment (carlpulley) [reply src/sbt-test/sbt-cake/simple/src/main/scala/Hello.scala:1]: Adding in an additional comment here feels like adding an unnecessary comment maintenance burden. So I'm with @fommil, let's keep this code as it currently is.

File: README.md [hunk 2]
```diff
@@ -91,6 +91,9 @@ The following SBT tasks are enabled:
 * `checkExternalBuildTools` - checks that all commands in `externalBuildTools` run without generating a non-zero exit
   code.
 
+In continuous integration (CI) environments, the SBT task `checkExternalBuildTools` should be ran by build scripts to
+ensure that all requiste build tools (e.g. `git`, `docker`, etc.) are installed.
```
Comment (ferhtaydn) [inline README.md]: `requiste` typo?

**Review by amirkarimi — APPROVED** [review]

**Pull request merged**